clients/client-python/tests/integration/test_gvfs_with_hdfs.py

```diff
@@ -0,0 +1,788 @@
+        try:
+            cls._clean_test_data()
+            # reset server conf
+            cls._reset_server_hadoop_conf()
+            # restart server
+            cls.restart_server()
+        finally:
+            # close hdfs container
+            cls.hdfs_container.close()
+
+    @classmethod
+    def _append_server_hadoop_conf(cls):
+        logger.info("Append server hadoop conf.")
+        gravitino_home = os.environ.get("GRAVITINO_HOME")
+        if gravitino_home is None:
+            raise GravitinoRuntimeException("Cannot find GRAVITINO_HOME env.")
+        hadoop_conf_path = f"{gravitino_home}/catalogs/hadoop/conf/hadoop.conf"
+        if not os.path.exists(hadoop_conf_path):
+            raise GravitinoRuntimeException(
+                f"Hadoop conf file is not found at `{hadoop_conf_path}`."
+            )
+        hdfs_container_ip = cls.hdfs_container.get_ip()
+        with open(hadoop_conf_path, mode="a", encoding="utf-8") as f:
+            key = "gravitino.bypass.fs.defaultFS"
+            value = f"hdfs://{hdfs_container_ip}:9000"
+            f.write(f"\n{key} = {value}")
+
+    @classmethod
+    def _reset_server_hadoop_conf(cls):
+        logger.info("Reset server hadoop conf.")
+        gravitino_home = os.environ.get("GRAVITINO_HOME")
+        if gravitino_home is None:
+            raise GravitinoRuntimeException("Cannot find GRAVITINO_HOME env.")
+        hadoop_conf_path = f"{gravitino_home}/catalogs/hadoop/conf/hadoop.conf"
+        if not os.path.exists(hadoop_conf_path):
+            raise GravitinoRuntimeException(
+                f"Hadoop conf file is not found at `{hadoop_conf_path}`."
+            )
+        with open(hadoop_conf_path, mode="r", encoding="utf-8") as file:
+            lines = file.readlines()
+        hdfs_container_ip = cls.hdfs_container.get_ip()
+        key = "gravitino.bypass.fs.defaultFS"
+        value = f"hdfs://{hdfs_container_ip}:9000"
+        key_to_delete = f"{key} = {value}"
+        filtered_lines = [line for line in lines if not line.startswith(key_to_delete)]
+        with open(hadoop_conf_path, mode="w", encoding="utf-8") as file:
+            file.writelines(filtered_lines)
+
+    @classmethod
+    def _download_and_unzip_hadoop_pack(cls):
+        logger.info("Download and unzip hadoop pack from %s.", HADOOP_DOWNLOAD_URL)
+        local_tar_path = f"{LOCAL_HADOOP_DIR}/{HADOOP_PACK_NAME}"
+        # will download from remote if we do not find the pack locally
+        if not os.path.exists(local_tar_path):
+            response = requests.get(HADOOP_DOWNLOAD_URL)
+            with open(local_tar_path, "wb") as f:
+                f.write(response.content)
+        # unzip the pack
+        try:
+            with tarfile.open(local_tar_path) as tar:
+                tar.extractall(path=LOCAL_HADOOP_DIR)
+        except Exception as e:
+            raise GravitinoRuntimeException(
+                f"Failed to extract file '{local_tar_path}': {e}"
+            ) from e
```

Comment:
Why do we need to download hadoop package?

Reply:
The reason is that Python integration tests is equivalent to the client. When we access HDFS through `Pyarrow.HadoopFileSystem` in `PyGVFS`, we need to use the `libhdfs` binary file in the Hadoop environment. Therefore, we need to download the Hadoop distribution package for the client and configure the Hadoop environment.

Reply:
Can we separate this into a base class or a helper class, don't mix everything in the unit tests, which is hard to review.

Reply:
Got it, I will refactor it into a Hadoop environment base class.